pp_netlib/network.py

```diff
@@ -235,10 +236,58 @@ def get_summary(self, print_to_std = True, summary_file_prefix = None):
                 summary.write(summary_contents)
             summary.close()
 
-    def visualize(self):
-        # code for calling viz functions
-        print("visualizing network")
-        return #files associated with viz
+    def visualize(self, viz_type, out_prefix, external_data = None):
+        from pp_netlib.functions import generate_mst_network, save_graph
+
+        self.mst_network = generate_mst_network(self.graph, self.backend)
+        
+        if viz_type == "mst":
+
+            mst_outdir = os.path.join(self.outdir, "mst")
+            if not os.path.exists(mst_outdir):
+                os.makedirs(mst_outdir)
+            file_name = out_prefix+"_mst_network_data"
+            save_graph(self.mst_network, backend = self.backend, outdir = mst_outdir, file_name = file_name, file_format = ".graphml")
+
+            if self.backend == "GT":
+                from pp_netlib.functions import draw_gt_mst, get_gt_clusters
+
+                isolate_clustering = get_gt_clusters(self.graph)
+                draw_gt_mst(mst = self.mst_network, out_prefix = os.path.join(mst_outdir, out_prefix), isolate_clustering=isolate_clustering, overwrite=True)
+            
+            if self.backend == "NX":
+                from pp_netlib.functions import draw_nx_mst, get_nx_clusters
+
+                isolate_clustering = get_nx_clusters(self.graph)
+                draw_nx_mst(mst=self.mst_network, out_prefix=os.path.join(mst_outdir, out_prefix), isolate_clustering=isolate_clustering, overwrite=True)
+
+        if viz_type == "cytoscape":
+            from pp_netlib.functions import write_cytoscape_csv
+        
+            cytoscape_outdir = os.path.join(self.outdir, "cytoscape")
+            if not os.path.exists(cytoscape_outdir):
+                os.makedirs(cytoscape_outdir)
+            save_graph(self.graph, self.backend, cytoscape_outdir, out_prefix+"_cytoscape", ".graphml")
+            save_graph(self.mst_network, self.backend, cytoscape_outdir, out_prefix+"_mst", ".graphml")
+
+            if self.backend == "GT":
+                from pp_netlib.functions import gt_save_graph_components, get_gt_clusters
+                gt_save_graph_components(self.graph, out_prefix, cytoscape_outdir)
+                clustering = get_gt_clusters(self.graph)
+
+                write_cytoscape_csv(os.path.join(cytoscape_outdir, out_prefix+".csv"), clustering.keys(), clustering, external_data)
+
+            if self.backend == "NX":
+                from pp_netlib.functions import nx_save_graph_components, get_nx_clusters
+                nx_save_graph_components(self.graph, out_prefix, cytoscape_outdir)
+                clustering = get_nx_clusters(self.graph)
+
+                write_cytoscape_csv(os.path.join(cytoscape_outdir, out_prefix+".csv"), clustering.keys(), clustering, external_data)
+
+            
+            # in progress
+            # self.write_metadata(cytoscape_outdir, out_prefix, external_data)
```

Comment:
What is the plan here?

Reply:
the idea was to get node data (node id, cluster), and then merge it with any external/epidemiological data associated with the samples and save as csv. This csv might be useful for cytoscape as well as other analysis

Reply:
Ok, can you remove the unused part from this PR for now (you can leave it on another branch) – we should talk about how much more we expect to do on this package before you leave

Reply:
sure, i'll also remove other unused portions from this pr as you've indicated